### PR TITLE
Mantine 7

### PR DESCRIPTION
**Pull request by dvmoritzschoefl**
Draft PR for Mantine 7 migration.

### CODE REVIEW

File: src/Globals.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+declare module '*.module.css';
+declare module '*.module.scss';
```
Comment (puehringer) [inline src/Globals.d.ts:2]: Move to tsd.d.ts

File: src/app/VisynApp.tsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { AppShell, AppShellProps } from '@mantine/core';
 import * as React from 'react';
 import { JSXElementConstructor, ReactElement } from 'react';
+import { useVisynAppContext } from './VisynAppContext';
 import { VisynHeader } from './header/VisynHeader';
 import { VisynLoginMenu } from './login/VisynLoginMenu';
-import { useVisynAppContext } from './VisynAppContext';
+
+import '@mantine/core/styles.css';
```
Comment (puehringer) [inline src/app/VisynApp.tsx:8]: Add other packages here as well. 

https://mantine.dev/styles/mantine-styles/#mantine-components-styles

File: src/demo/MainApp.tsx [hunk 3]
```diff
@@ -82,6 +84,18 @@ export function MainApp() {
...
+          <Box
+            style={
+              {
+                '--my-hover': 'red',
+              } as React.CSSProperties
+            }
+            className={classes.test}
+          >
+            Test
+          </Box>
```
Comment (puehringer) [inline src/demo/MainApp.tsx:97]: Remove

**Comment by dvmoritzschoefl** [comment]
General things done in the migration:

- Replace `position="apart"` with `justify="space-between"` on all Stack/Group
- Replace `spacing=` with `gap=` on all Stack/Group (may also occur elsewhere)
- Replace `position="right"` with `justify="flex-end"` on all Stack/Group (may also occur in Tooltip)
- Replace `noWrap` with `wrap="nowrap"` on all Stack/Group
- Replace `position="center"` with `justify="center"` on all Stack/Group
- Search for `compact` and replace it with `size="compact-sm"` or the appropriate size (sm is default)
  - do this manually and check if there are already size attributes and change to correct size
- Replace `align=` with `ta=` on all Text
- Replace `sx` with `style` on all elements, but make sure
  - There are no inline selectors, hover styles etc
  - If there are, try to migrate them using css modules or emotion css
- Replace `weight="bold"` with `fw="bold"` on all Text (and all other weight variations)
- Remove `withinPortal` on all Elements
  - withinPortal is true by default now and not needed for most elements like Tooltip, only leave it if it says withinPortal="false"

**Pull request closed without merging**